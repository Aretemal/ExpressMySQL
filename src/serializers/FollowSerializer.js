import Serializer from './Serializer.js';

class FollowSerializer extends Serializer {
  type() {
    return 'Follow';
  }

  attributes() {
    return {
      followerId: this.resource.attributes.followerId,
      followingId: this.resource.attributes.followingId,
      approvedAt: this.resource.attributes.approvedAt,
    };
  }

  id() {
    return this.resource.attributes.id || 0;
  }
}
export default FollowSerializer;
