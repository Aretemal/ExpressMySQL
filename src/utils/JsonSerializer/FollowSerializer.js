import Serializer from './Serializer.js';

class FollowSerializer extends Serializer {
  follow(data, links) {
    return this.serializer(data, data.id, 'Follow connection', links);
  }

  approve(data, links) {
    return this.serializer(data, data.id, 'Approve connection', links);
  }

  unfollow(data, links) {
    return this.serializer(data, data.id || 0, 'Unfollow connection', links);
  }
}
export default new FollowSerializer();
