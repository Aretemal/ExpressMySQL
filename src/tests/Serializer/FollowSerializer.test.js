import FollowSerializer from '../../utils/JsonSerializer/FollowSerializer.js';

describe('FollowSerializer :', () => {
  describe('Follow :', () => {
    test('should return connection with info', () => {
      const links = 'http://localhost:5000/api/follow';
      const connection = {
        id: 21, followerId: 1, followingId: 2, approvedAt: 0,
      };

      const data = FollowSerializer.follow(connection, links);

      expect(data).toMatchSnapshot();
    });
  });

  describe('Approve :', () => {
    test('should return connection with info', () => {
      const links = 'http://localhost:5000/api/approve';
      const connection = {
        id: 21, followerId: 1, followingId: 2, approvedAt: 1,
      };

      const data = FollowSerializer.approve(connection, links);

      expect(data).toMatchSnapshot();
    });
  });

  describe('Unfollow :', () => {
    test('should return connection with info', () => {
      const links = 'http://localhost:5000/api/unfollow';
      const connection = {
        id: 21, followerId: 1, followingId: 2, approvedAt: 0,
      };

      const data = FollowSerializer.unfollow(connection, links);

      expect(data).toMatchSnapshot();
    });
  });
});
