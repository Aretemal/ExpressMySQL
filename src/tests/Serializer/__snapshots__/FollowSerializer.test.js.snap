// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowSerializers : Approve : should return connection with info 1`] = `
{
  "data": {
    "attributes": {
      "approvedAt": 1,
      "followerId": 1,
      "followingId": 2,
      "id": 21,
    },
    "id": 21,
    "type": "Approve connection",
  },
  "links": {
    "self": "http://localhost:5000/api/approve",
  },
}
`;

exports[`FollowSerializers : Follow : should return connection with info 1`] = `
{
  "data": {
    "attributes": {
      "approvedAt": 0,
      "followerId": 1,
      "followingId": 2,
      "id": 21,
    },
    "id": 21,
    "type": "Follow connection",
  },
  "links": {
    "self": "http://localhost:5000/api/follow",
  },
}
`;

exports[`FollowSerializers : Unfollow : should return connection with info 1`] = `
{
  "data": {
    "attributes": {
      "approvedAt": 0,
      "followerId": 1,
      "followingId": 2,
      "id": 21,
    },
    "id": 21,
    "type": "Unfollow connection",
  },
  "links": {
    "self": "http://localhost:5000/api/unfollow",
  },
}
`;
